Functional_hw/src/de/telran/Main.java

```diff
@@ -0,0 +1,32 @@
+package de.telran;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class Main {
+
+    public static void main(String [] args){
+
+        Address address1 = new Address("Berliner Str.", 29);
+        Address address2 = new Address("Akazien Str.", 22);
+
+        Person ivan = new Person("Ivan", 35, address1);
+        Person alex = new Person("Alex", 16, address2);
+
+        ArrayList<Person> persons = new ArrayList<>(Arrays.asList (ivan, alex));
+        getAddressFromPersonMore17(persons);
+
+
+
+    }
+
+    public static List<Address> getAddressFromPersonMore17(List <Person> persons) {
+
+        return (List<Address>) persons.stream()
```

Comment:
You do not need type casting:  `(List<Address>)` since the resulting list should already be of the list with addresses.